api/src/pcapi/connectors/beamer.py

```diff
@@ -0,0 +1,92 @@
+import logging
+
+from pcapi import settings
+from pcapi.core.external.attributes import models as attributes_models
+from pcapi.utils import requests
+from pcapi.utils.module_loading import import_string
+
+
+logger = logging.getLogger(__name__)
+
+
+def update_beamer_user(pro_attributes: attributes_models.ProAttributes) -> None:
+    _get_backend().update_pro_user(pro_attributes)
+
+
+def _get_backend() -> "BaseBackend":
+    backend_class = import_string(settings.BEAMER_BACKEND)
+    return backend_class()
+
+
+class BaseBackend:
+    url = "https://api.getbeamer.com/v0"
+    api_key = settings.BEAMER_API_KEY
+
+    def update_pro_user(self, pro_attributes: attributes_models.ProAttributes) -> None:
+        """Upserts the user into the Beamer database"""
+        raise NotImplementedError()
+
+
+class BeamerBackend(BaseBackend):
+    def update_pro_user(self, pro_attributes: attributes_models.ProAttributes) -> None:
+        """Upserts the user into the Beamer database"""
+        if not pro_attributes.user_id:
+            raise MissingIdException()
+
+        put_users_url = f"{self.url}/users"
+        try:
+            response = requests.put(
+                put_users_url,
+                headers={"Beamer-Api-Key": self.api_key},
+                json=format_pro_attributes(pro_attributes),
+            )
+        except requests.exceptions.RequestException as exc:
+            logger.exception(
+                "Network error on Adresse API",
+                extra={"exc": exc, "url": put_users_url, "userId": pro_attributes.user_id},
+            )
+            raise BeamerNetworkException from exc
+
+        if response.status_code != 200:
+            raise BeamerException(
+                f"Unexpected {response.status_code} response from Beamer for user {pro_attributes.user_id}"
+            )
+
+
+class LoggerBackend(BaseBackend):
+    def update_pro_user(self, pro_attributes: attributes_models.ProAttributes) -> None:
+        if not pro_attributes.user_id:
+            raise MissingIdException()
+        request_data = format_pro_attributes(pro_attributes)
+        logger.info("PUT %s/users with data = %s", self.url, request_data)
```

Comment:
(détail / question de goût) J'aurais mis quelque chose de moins technique (moins lié à l'implémentation de `BeamerBackend`, qui ne nous intéresse pas vraiment ici), du genre : 

```suggestion
        logger.info("Updated pro user data on Beamer: %s", request_data)
```

... ce qui permettrait de déplacer les attributs de class `url` et `api_key` de `BaseBackend` vers `BeamerBackend` (qui est le seul à utiliser ces valeurs).